configure.go

```diff
@@ -0,0 +1,144 @@
+package configure
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"reflect"
+	"strings"
+)
+
+const (
+	structTagKey             = "config"
+	requiredTagKey           = "required"
+	missingValuesErrTemplate = "Missing required fields: %s"
+)
+
+var (
+	ErrStringsOnly            = errors.New("Only string values are allowed in a config struct.")
+	ErrNotReference           = errors.New("The config struct must be a pointer.")
+	ErrStructOnly             = errors.New("Config object must be a struct.")
+	ErrNoTagValue             = errors.New("Config object attributes must have a 'config' tag value.")
+	ErrFlagParsed             = errors.New("The flag library cannot be used in conjunction with configure")
+	ErrInvalidJSON            = errors.New("Invalid JSON found in arguments.")
+	ErrStructTagInvalidOption = errors.New("Only 'required' is a config option.")
+)
+
+func parseTagKey(tag string) (key string, required bool, err error) {
+	s := strings.Split(tag, ",")
+	switch len(s) {
+	case 2:
+		if s[1] != requiredTagKey {
+			return "", false, ErrStructTagInvalidOption
+		}
+		return s[0], true, nil
+	case 1:
+		return s[0], false, nil
+	default:
```

Comment:
wait, nvm. `strings.Split` always returns a list of at least 1, it will insert an empty string if provided with an empty string